cmd/main.go

```diff
@@ -4,24 +4,31 @@ import (
 	"bufio"
 	"flag"
 	"fmt"
-	"log"
+	"log/slog"
 	"os"
 	"time"
 
 	"github.com/kevmo314/appendable/pkg/appendable"
 )
 
 func main() {
+	var debugFlag, jsonlFlag, csvFlag, showTimings bool
 
-	var jsonlFlag bool
-	var csvFlag bool
-
+	flag.BoolVar(&debugFlag, "debug", false, "Use logger that prints at the debug-level")
 	flag.BoolVar(&jsonlFlag, "jsonl", false, "Use JSONL handler")
 	flag.BoolVar(&csvFlag, "csv", false, "Use CSV handler")
-
-	var showTimings bool
 	flag.BoolVar(&showTimings, "t", false, "Show time-related metrics")
 
+	flag.Parse()
+
+	var logLevel = new(slog.LevelVar)
+
+	if debugFlag {
+		logLevel.Set(slog.LevelDebug)
+	}
+
+	var logger = slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: logLevel}))
+
```

Comment:
You can call `slog.SetDefault(logger)` to avoid needing to pass the logger around and instead call `slog.Info()`, etc.